site2/docs/functions-develop.md

```diff
@@ -0,0 +1,570 @@
+---
+id: functions-develop
+title: Develop Pulsar Functions
+sidebar_label: Develop functions
+---
+
+This tutorial walks you through how to develop Pulsar Functions.
+
+## Available APIs
+In Java, Python, and Go, you have two options to write Pulsar Functions.
+
+Interface | Description | Use cases
+:---------|:------------|:---------
+Language-native interface | No Pulsar-specific libraries or special dependencies required (only core libraries from Java/Python/Go). | Functions that do not require access to the function [context](#context).
+Pulsar Function SDK for Java/Python/Go | Pulsar-specific libraries that provide a range of functionality not provided by "native" interfaces. | Functions that require access to the function [context](#context).
+
+<!--DOCUSAURUS_CODE_TABS-->
```

Comment:
Can we have a separate section for examples using the SDK API.  I don't think its the best to coalesce language native and SDK examples together

Reply:
If we can also provide links to the complete code, I think that would also be help to the reader

Reply:
Sure, separate them.